src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
+}
 
+function round(num: number): number {
+  return Math.floor(num * 100) / 100;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const maxSide = Math.max(a, b, c);
+    const sumOfLowSides = [a, b, c]
+      .filter((side) => side !== maxSide)
+      .reduce((sum, side) => sum + side);
 
-export class Circle {
+    if (a <= 0 || b <= 0 || c <= 0 || maxSide >= sumOfLowSides) {
+      throw new Error(
+        'Incorrect data: impossible to build a triangle with this values',
+      );
+    }
```

Comment:
You check that input values are bigger than 0 for each class
Will be better to create separate function, that will accept initial parameters, check it and return the corresponding error if not
And you can call this one general function in each class